core/src/main/java/org/apache/druid/java/util/common/guava/ParallelMergeCombiningSequence.java

```diff
@@ -341,6 +344,7 @@ protected void compute()
         }
       }
       catch (Exception ex) {
+        closeAllCursors(sequenceCursors);
```

Comment:
Sorry, I misread this. A finally block won't work 🤦 

Does it provide any better guarantees if we catch a Throwable instead of an exception here - so that in case any of the code in the try block throws an `Error` instead of an `Exception`

This code appears to have been running ok all this time with a `catch (Exception)` block so I'm guessing it's not too bad if the catch condition is left as is

Reply:
Seems reasonable to switch to catching throwable, can swap it out